python/pyspark/sql/functions.py

```diff
@@ -116,7 +114,7 @@ def __init__(self, func, returnType):
 
     def _create_judf(self):
         f = self.func  # put it in closure `func`
-        func = lambda _, it: imap(lambda x: f(*x), it)
+        func = lambda _, it: map(lambda x: f(*x), it)
```

Comment:
A common approach I've seen in projects wanting to support both Python 2 and 3 is to use the [`six`](https://pythonhosted.org/six/) compatibility module, which has [support for renamed methods](https://pythonhosted.org/six/#module-six.moves).

```
from six.moves import map
```

We probably don't want to add another external dependency, but just thought I'd throw that out there.